lib/backend/dynamo/dynamodbbk.go

```diff
@@ -283,19 +285,52 @@ func New(ctx context.Context, params backend.Params) (*Backend, error) {
 
 	otelaws.AppendMiddlewares(&awsConfig.APIOptions, otelaws.WithAttributeSetter(otelaws.DynamoDBAttributeSetter))
 
+	dynamoClient := dynamodb.NewFromConfig(awsConfig, dynamoOpts...)
+	streamsClient := dynamodbstreams.NewFromConfig(awsConfig)
 	b := &Backend{
 		logger:  l,
 		Config:  *cfg,
 		clock:   clockwork.NewRealClock(),
 		buf:     backend.NewCircularBuffer(backend.BufferCapacity(cfg.BufferSize)),
-		svc:     dynamodb.NewFromConfig(awsConfig, dynamoOpts...),
-		streams: dynamodbstreams.NewFromConfig(awsConfig),
+		svc:     dynamoClient,
+		streams: streamsClient,
 	}
 
 	if err := b.configureTable(ctx, applicationautoscaling.NewFromConfig(awsConfig)); err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	// Perform an initial endpoint resolution to retrieve the URI of the DynamoDB services
+	// so that it can be included in the logging output below.
+	var dynamoEndpoint url.URL
+	clientOpts := dynamoClient.Options()
+	resolved, err := clientOpts.EndpointResolverV2.ResolveEndpoint(ctx, dynamodb.EndpointParameters{
+		Region:   aws.String(clientOpts.Region),
+		UseFIPS:  aws.Bool(clientOpts.EndpointOptions.UseFIPSEndpoint == aws.FIPSEndpointStateEnabled),
+		Endpoint: clientOpts.BaseEndpoint,
+	})
+	if err == nil {
```

Comment:
Should we log the error if we failed to resolve? Otherwise the log entry will contain empty URL and it might not be clear why.